api.md

```diff
@@ -266,71 +266,157 @@ Output:
 
 Status Code: 200 or 400
 
+### Create Project
+
+POST "/project_create"
+
+This will create a project with the current user as the admin.
+
+Only requires name and description of the project.
+
+Input: A JSON body with the following **required** parameters.
+
+```typescript
+{
+    name: string;
+    description: string;
+}
+```
+
+Status Code: 201 or 400
+
+### Get Project
+
+GET "/project_get"
+
+This will get the project's name, description and creation time.
+
+Input: A JSON body with the following **required** parameters.
+
+```typescript
+{
+    projectid: string;
+}
+```
+
+Status Code: 200 or 400
+
+### Create Task
+
+POST "/task_create"
+
+This will create either a personal task or project task based on if a project ID is passed in.
+No projectid passed in, the task will be created for current user.
+If specified, will create a task for the project and assign users in user array to that task.
+
+Input: A JSON body with the following **required** parameters.
+
+```typescript
+{
+  name: string;
+  description: string;
+  users: string[];
+  projectID: string;
+}
+```
+
+### Add User to Task
+
+PATCH "/task_add_user"
```

Comment:
-> we need a task_patch/task_modify as well (potentially merge with task_add_user or maybe just keep them separate)
-> this is used for both task editing and task drag & dropping (probably have to add dnd_id into Task model on server side as well)